### DISCUSSION TITLE
Ccache not being used at all on Windows with intel compilers

### DISCUSSION THREAD

**Discussion by Yattabyte** [ccache/ccache#1337] [Support]
### How to reproduce ###

1. Use windows 10
2. Use latest Intel OneAPI's Next-Gen DPC/C++ Compilers (ICC is discontinued, ICX is the only one shipped now)
3. Use CMake and set `CMAKE_CXX_COMPILER_LAUNCHER <path to ccache> <options>`
4. export CC to ICX, and CXX to ICPX
5. Configure a C++ project with cmake using the `-T "Intel C++ Compiler 2023"` toolset,
6. Build, and confirm the cache size unchanged

### Actual behavior ###
Ccache appears on the surface to be completely bypassed, inexplicably. Using the `log_file` option I don't get an ounce of output even. Checking the stats with `-s`, nothing other than the cache size of 0/20 gb is outputted.

### Expected behavior ###
Using identical CMake code, on Mac and Linux (where I also use the same compiler), I'd expect ccache to correctly cache the compiled object files, and when running `-s` see the stats of hits and misses.

### Environment ###
Using windows 10 64 bit (likely latest version), cmake 3.23.3, ccache 4.8.3. MSBuild instead of Ninja (for now). Git-bash is used on windows instead of powershell/cmd, with a few cygwin utilities to maintain a similar bash environment between mac/linux/windows.

The specific CMake code I use is:
```cmake
find_program(CCACHE_PROGRAM ccache)
if (CCACHE_PROGRAM)
    message(NOTICE "Selected CCache: ${CCACHE_PROGRAM}")
    set(CMAKE_C_COMPILER_LAUNCHER ${CCACHE_PROGRAM} base_dir=${PROJECT_SOURCE_DIR} hash_dir=false log_file=${PROJECT_SOURCE_DIR}/tmp_ccache.log)
    set(CMAKE_CXX_COMPILER_LAUNCHER ${CCACHE_PROGRAM} base_dir=${PROJECT_SOURCE_DIR} hash_dir=false log_file=${PROJECT_SOURCE_DIR}/tmp_ccache.log)
else ()
    message(WARNING "Could not find ccache, skipping!")
endif ()
```
The intention behind `base_dir` and `hash_dir` in this context is that on my other mac/linux servers the cache is shared and projects are not checked out at the same path all the time between different accounts/jobs. Doesn't matter on windows yet since the cache is never populated it seems.
Just to make it clear, cmake _**does find ccache**_

### Other Notes ###
For what its worth, the Intel compilers are now clang based. They seem to work identically between linux and windows now. I've tried setting CCACHE_CC/CXX to ICX/ICPX, and also tried setting `CCACHE_COMPILERTYPE` to clang/clang-cl/icl to no avail.

**Comment by cristianadam** [comment]
If you have a look at https://cmake.org/cmake/help/latest/prop_tgt/LANG_COMPILER_LAUNCHER.html you will see this part:

> Specify a [semicolon-separated list](https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#cmake-language-lists) containing a command line for a compiler launching tool. The [Makefile Generators](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#makefile-generators) and the [Ninja](https://cmake.org/cmake/help/latest/generator/Ninja.html#generator:Ninja) generator will run this tool and pass the compiler and its arguments to the tool

CMake doesn't use the compiler launcher for the MSBuild generator.

Have a look at https://github.com/ccache/ccache/wiki/MS-Visual-Studio#usage-with-cmake to find out how to use ccache with msbuild.

**Comment by Yattabyte** [comment]
Ah I missed that in the documentation. Thanks, I'll give that a shot!

**Reply by vryy** [reply]
@Yattabyte were you successful compiling using Intel Compiler with ccache on MSVisualStudio? If yes, can you share the CMake setup?
